### ISSUE TITLE
[solc] CompilerContext.cpp throws ICE "Invalid stack height in low-level function" because the check "stackHeight() == outArgs" fails

### ISSUE THREAD

**Issue by bshastry** [ethereum/solidity#8448]
## Description

```
pragma experimental ABIEncoderV2;

contract C {
  uint[][2] tmp_i;
  function i(uint[][2] calldata s) external { tmp_i = s; }
}
```

throws

https://github.com/ethereum/solidity/blob/a328e940b5351e1411537da74485a112d5dad608/libsolidity/codegen/CompilerContext.cpp#L146

## Environment

- Compiler version: latest develop
- Target EVM version (as per compiler settings): NA

## Steps to Reproduce

```
$ solc <input.sol>
Internal compiler error during compilation:
/home/bhargava/work/github/solidity/libsolidity/codegen/CompilerContext.cpp(146): Throw in function void solidity::frontend::CompilerContext::appendMissingLowLevelFunctions()
Dynamic exception type: boost::exception_detail::clone_impl<solidity::langutil::InternalCompilerError>
std::exception::what: Invalid stack height in low-level function $copyArrayToStorage_t_array$_t_array$_t_uint256_$dyn_calldata_$2_calldata_ptr_to_t_array$_t_array$_t_uint256_$dyn_storage_$2_storage.
[solidity::util::tag_comment*] = Invalid stack height in low-level function $copyArrayToStorage_t_array$_t_array$_t_uint256_$dyn_calldata_$2_calldata_ptr_to_t_array$_t_array$_t_uint256_$dyn_storage_$2_storage.
```

**Comment by axic** [comment]
Still a problem.

**Comment by Marenz** [comment]
Closed via #7635
